src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -326,6 +326,20 @@ public function testGetMimeTypeFromFormat($format, $mimeTypes)
         }
     }
 
+    /**
+     * @dataProvider getFormatToMimeTypeMapProviderForAccept
+     */
+    public function testGetMimeTypeFromFormatWithAccept($format, $mimeTypes, $accepts)
+    {
+        if (null !== $format) {
+            $request = new Request();
+            $request->headers->set('Accept', $accepts);
+            $request->setFormat($format, $mimeTypes);
+            //Only when accept value matches a value verbatim inside of the mimetype array.
+            $this->assertEquals($accepts, $request->getMimeType($format));
```

Comment:
Wouldn't the same thing happen now as the code currently exists?

If I ran the test you're describing against the codebase as it exists right now, I would get the same result, or perhaps I am mistaken?

The situation is that the real use of getMimeType in the stack will always match the format that the request has, and it currently does not honor mime_type versions.

Unless you think we should couple the getMimeType return value to the accept header no matter what (in which case then the `static::$formats[$format][0];` still seems wrong?

Thoughts?